api/v1beta1/common.go

```diff
@@ -60,6 +65,26 @@ func validateIBMPowerVSResourceReference(res IBMPowerVSResourceReference, resTyp
 	return true, nil
 }
 
+func validateIBMPowerVSMemoryValues(resValue string) bool {
+	if val, err := strconv.ParseUint(resValue, 10, 64); err != nil || val == 0 {
+		return false
+	}
+	return true
+}
+
+func validateIBMPowerVSProcessorValues(resValue string) bool {
+	valueTypes := [...]string{"nan", "inf"}
+	for _, vt := range valueTypes {
+		if strings.Contains(strings.ToLower(resValue), vt) {
+			return false
+		}
+	}
+	if val, err := strconv.ParseFloat(resValue, 64); err != nil || val <= 0 {
+		return false
+	}
+	return true
+}
+
```

Comment:
This doesn't look clean, wondering if we can use regex marker for this one `+kubebuilder:validation:Pattern`, one another point this allows like -12.345 as well, which is not a valid number.